ROOTFS/etc/waggle/sanity/fatal/static_psu_control.test

```diff
@@ -0,0 +1,159 @@
+        fi
+        if [ -d $gpiopath/gpio$pg4 ]; then
+                echo $pg4 > $gpiopath/unexport
+        fi
+}
+trap gpio_cleanup EXIT
+
+
+# Not my code, taken from Joe's PSU test
+
+# export the GPIOs and set to input
+if [ ! -d $gpiopath/gpio$acok ]; then
+        echo $acok > $gpiopath/export
+fi
+echo "in" > $gpiopath/gpio$acok/direction
+if [ ! -d $gpiopath/gpio$pg1 ]; then
+        echo $pg1 > $gpiopath/export
+fi
+echo "in" > $gpiopath/gpio$pg1/direction
+if [ ! -d $gpiopath/gpio$pg2 ]; then
+        echo $pg2 > $gpiopath/export
+fi
+echo "in" > $gpiopath/gpio$pg2/direction
+if [ ! -d $gpiopath/gpio$pg3 ]; then
+        echo $pg3 > $gpiopath/export
+fi
+echo "in" > $gpiopath/gpio$pg3/direction
+if [ ! -d $gpiopath/gpio$pg4 ]; then
+        echo $pg4 > $gpiopath/export
+fi
+echo "in" > $gpiopath/gpio$pg4/direction
+
+
+# PG2 is the Edge processor, and should correspond to whether it is active in the manifest
+isEPPresent=$(cat /etc/waggle/node_manifest.json | jq .psu.ep_supply.present)
+
+# PG3/PG4 are bound, and should be treated as one AND'ed value. If they aren't AND'ed, big trouble. This AND'ed value should correspond to the peripherals in the manifest.
+isPHPresent=$(cat /etc/waggle/node_manifest.json | jq .psu.ph_supply.present )
+
+# check that we isEPPresent and isPHPresent have values
+
+if [[ $isPHPresent == "" ]];
+then 
+        echo "$pf: Could not get state of PH from manifest FAIL"
+fi
+
+if [[ $isEPPresent == "" ]];
+then 
+        echo "$pf: Could not get state of EP from manifest FAIL"
+fi
+
+
+
+echo "Node EP availability: $isEPPresent"
+echo "Node PH availability: $isPHPresent"
+
+
+bool_to_int() { 
+
+        if $1 -eq "true";
+        then echo "1"
+        else 
+        echo "0"
+        fi
+
+}
+
+
+# PG1 and ACOK Should always be high
+# Error codes 45 and 46 are respectively for failure of ACOK and PG1 
+# Error code 47 is for misaligned EP and PG2
+# Error code 48 is for misaligned PG3 and PG4
+# Error code 49 is for misaligned EP and PG3/PG4
+
+if ! gpio_high $acok; then
+        echo "$pf: ACOK3 Not powered FAIL"
+        exit 45
+fi
+
+if ! gpio_high $pg1; then
+        echo "$pf: PG1  Not powered FAIL"
+        exit 46
+fi
+
+
+# comparing EPSupply
+
+if [[  $(bool_to_int $isEPPresent) ==  $(gpio_high $pg2) ]];
+then 
+        echo "$pf: EP is $(bool_to_int $isEPPresent) in manifest, but PG2 is set to $? FAIL"
+        exit 47
+fi
+
+# make sure both pg3 and pg4 are the same
+if [[ $(gpio_high $pg3) != $(gpio_high $pg3) ]]; 
+then 
+        echo "$pf: PG3 and PG4 are not aligned FAIL"
+        exit 48
+fi
+
+# check that either of them align with PH. In this case, I'll just use pg3 
+# like the previous comparison, I copied the gpio_high function from joe, so a value of high will be 0, low will be 1. 
+# but a value of true for PH is 1, and a value of false is 0, so if they're equal it means one is high and the other is low.
+
+if [[ $(bool_to_int $isPHPresent) == $(gpio_high $pg3)  ]];
```

Comment:
while this works (as does the test above for `isEPPresent`) it is a little hard to read.  we can talk about this but i think this is a little easier to read

```
if isPHPresent
  if ! gpio_high pg3
     error condition 1
  if ! gpio_high pg4
     error condition 2
```

same idea for the `isEPPresent` test.

also now that I think about it, it would be valuable to print the value of all the GPIOs at the start of this test and then test that they are the right values.  That way if there is an error, we can go back in the logs and see what the value of _ALL_ the gpios were during this error condition.  My above example (if pg3 was wrong) would never test and therefore print the value of pg4.